includes/wikia/services/UserStatsService.class.php

```diff
@@ -182,159 +154,91 @@ public function getEditCountGlobal( $wikiId = 0, $skipCache = false ) {
-
-			$this->optionsAllWikis[ $wikiId ] = $this->wg->Memc->get( $key );
-
-			if ( !empty( $this->optionsAllWikis[ $wikiId ] ) ) {
-				wfProfileOut( __METHOD__ );
-				return $this->optionsAllWikis[ $wikiId ];
-			}
-
-		}
+	public function getEditCountFromWeek( $skipCache = false ) {
+		wfProfileIn( __METHOD__ );
 
-		$dbName = ( $wikiId == $this->wg->CityId ) ? false : WikiFactory::IDtoDB( $wikiId );
+		$editCount = $this->getOptionWiki( 'editcountThisWeek', $skipCache );
 
-		/* Get option value from wiki specific user properties */
-		$dbr = $this->getWikiDB( DB_SLAVE, $dbName );
-		$res = $dbr->select(
-			'wikia_user_properties',
-			array ( 'wup_property', 'wup_value' ),
-			array(
-				'wup_user' => $this->userId
-			),
-			__METHOD__
-		);
-		$this->optionsAllWikis[ $wikiId ] = array();
-		foreach( $res as $row ) {
-			$this->optionsAllWikis[ $wikiId ][ $row->wup_property ] = $row->wup_value;
+		if( $editCount === null || $editCount === false ) { // editcount has not been initialized. do so.
+			$editCount = $this->calculateEditCountFromWeek();
 		}
-		$this->wg->Memc->set( $key, $this->optionsAllWikis[ $wikiId ], self::CACHE_TTL );
 
 		wfProfileOut( __METHOD__ );
-		return $this->optionsAllWikis[ $wikiId ];
+		return (int)$editCount;
 	}
 
-
 	/**
-	 * Update localized value of editcount in wikia_user_properties
+	 * Update localized value of editcount and editcountThisWeek in wikia_user_properties
 	 * and bump mcached values for stats and localized user options
 	 */
-	function increaseEditsCount() {
-		wfProfileIn(__METHOD__);
+	public function increaseEditsCount() {
+		wfProfileIn( __METHOD__ );
 
 		// update edit counts in stats
-		$this->wg->Memc->delete( $this->getKey('stats4') );
+		self::purgeStatsCache( $this->userId, $this->getWikiId() );
 
+		// update edit counts on wiki
 		$editCount = $this->getOptionWiki( 'editcount' );
 		if ( !is_null( $editCount ) ) {
-			//update edit counts in options
-			$dbw = wfGetDB( DB_MASTER );
-			$dbw->update(
-				'wikia_user_properties',
-				[ 'wup_value=wup_value+1' ],
-				[ 'wup_user' => $this->userId, 'wup_property' => 'editcount' ],
-				__METHOD__
-			);
-
-			if ( $dbw->affectedRows() === 1 ) {
-				//increment memcache also
-				$key = wfSharedMemcKey( 'optionsWiki', $this->wg->CityId, $this->userId );
-				$optionsWiki = $this->wg->Memc->get( $key );
+			$this->updateEditCount( 'editcount' );
+		} else {
+			$this->resetEditCountWiki( Title::GAID_FOR_UPDATE );
+		}
 
-				$optionsWiki['editcount']++;
-				$editCount++;
-				$this->wg->Memc->set( $key, $optionsWiki, self::CACHE_TTL );
-			}
+		// update weekly edit counts on wiki
+		$editCount = $this->getOptionWiki( 'editcountThisWeek' );
+		if ( !is_null( $editCount ) ) {
+			$this->updateEditCount( 'editcountThisWeek' );
 		} else {
-			//initialize editcount skipping memcache
-			$editCount = $this->getEditCountWiki( 0, true, Title::GAID_FOR_UPDATE );
+			$this->calculateEditCountFromWeek( Title::GAID_FOR_UPDATE );
```

Comment:
If our system is hermetic, which should be the case this should always return 1. I suggest removing it.